### PR TITLE
Add decomposition and test for RMSNorm

### PR DESCRIPTION
**Pull request by avery-blanchard**
Thanks for sending a pull request! Please make sure you read the [contributing guidelines](https://github.com/torch-spyre/torch-spyre/blob/main/CONTRIBUTING.md).

#### What type of PR is this?

- [ ] bug
- [x] feature
- [ ] documentation
- [ ] cleanup

#### What this PR does:

This PR add a decomposition and test for RMSNorm.

### CODE REVIEW

File: torch_spyre/_inductor/decompositions.py [hunk 1]
```diff
@@ -37,6 +37,23 @@ def layernorm_decomp(
...
 
 
+@register_decomposition([torch.ops.spyre.rms_norm])
+def rmsnorm_decomp(
+    input: torch.Tensor,
+    normalized_shape: list[int],
+    weight: Optional[torch.Tensor] = None,
+    eps: float = 1e-5,
+) -> torch.Tensor:
+    # TODO: limitation with mean on dim=-1, transpose for now to avoid
```
[the remark this answers is not in the document]
Comment (ani300) [reply torch_spyre/_inductor/decompositions.py]: so we know to remove the fix when we can

File: tests/_inductor/test_inductor_ops.py [hunk 2]
```diff
@@ -986,6 +993,13 @@ def fn(q, k, v, sm_scale):
 
         compare_with_cpu(fn, *args)
 
+    @pytest.mark.filterwarnings("ignore::torch_spyre.fallbacks.FallbackWarning")
```
[the remark this answers is not in the document]
Comment (avery-blanchard) [reply tests/_inductor/test_inductor_ops.py]: `torch.ops.spyre.full` is used for eps [here](https://github.com/torch-spyre/torch-spyre/pull/645/changes#diff-11d87f25744b29772914fdaaaca93f22f94153b17f541201d422fe6e93ae7659R49) which fallbacks back to the CPU to create the tensor then copies it to the device.

File: torch_spyre/_inductor/decompositions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (avery-blanchard) [reply torch_spyre/_inductor/decompositions.py]: #637 
There is an issue with using output of torch.mean on dim=-1 in other operations. I am looking into this now, but did not want to block this. I will add more details to the comment.

[the remark this answers is not in the document]
Comment (ani300) [reply torch_spyre/_inductor/decompositions.py]: sure, just a link to the issue you just posted in the code should be enough for now!

**Review by ani300 — APPROVED** [review]
lgtm once torch.mean issue is linked in the code comments

**Review by ani300 — DISMISSED** [review]
lgtm

**Review by ani300 — DISMISSED** [review]
lgtm

**Review by ani300 — APPROVED** [review]
lgtm!
